core/modules/widgets/subclasses/edit-text-subclass.js

```diff
@@ -0,0 +1,65 @@
+/*\
+title: $:/core/modules/widgets/subclasses/edit-text-subclass.js
+type: application/javascript
+module-type: widget-subclass
+
+A subclass that intercepts the refresh method of edit-text widgets for enhanced refreshing
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+exports.baseClass = "edit-text";
+
+exports.constructor = function(parseTreeNode,options) {
+	this.initialise(parseTreeNode,options);
+};
+
+exports.prototype = {};
+
+// Configuration tiddlers
+var HEIGHT_MODE_TITLE = "$:/config/TextEditor/EditorHeight/Mode";
+var ENABLE_TOOLBAR_TITLE = "$:/config/TextEditor/EnableToolbar";
+
+exports.prototype.execute = function(event) {
+	// Call the base class execute function
+	Object.getPrototypeOf(Object.getPrototypeOf(this)).execute.call(this,event);
+	this.refreshTiddler = this.getAttribute("refreshTiddler");
+	this.refreshAction = this.getAttribute("refreshAction");
+	this.saveTiddler = this.getAttribute("saveTiddler");
+};
+
+exports.prototype.refresh = function(event) {
+	var changedAttributes = this.computeAttributes();
```

Comment:
IMO if you'd use 

```
var cA = this.computeAttributes();
```

The `else if (...` would be much easier to read.

Reply:
imo this is compliant with the core